### ISSUE TITLE
Support arm64 Docker Image

### ISSUE THREAD

**Issue by tnightengale** [unitycatalog/unitycatalog#480]
**Describe the bug**

Building images on M* Macs (arm64) doesn't seem to work. As far as I can tell, this is because of the choice to use alpine distro, which does not offer an arm64 compatible image (for jdk-17 only, sigh).

See [here](https://hub.docker.com/r/arm64v8/eclipse-temurin/) for the available eclipse-temurin arm64 tags.

Note the lack of an alpine tag for jdk-17 on arm64: 
<img width="1512" alt="Screenshot 2024-09-13 at 10 07 37 AM (2)" src="https://github.com/user-attachments/assets/c1c8ca72-8091-4a38-94b2-05636d954fd9">

**To Reproduce**

```
tnightengale@Teghans-14-MBP bin % ./build-uc-server-docker   
Changing directory to /Users/tnightengale/git/unitycatalog
Building unitycatalog:0.2.0-SNAPSHOT using the build definition from Dockerfile
#0 building with "desktop-linux" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 4.69kB done
#1 DONE 0.0s

#2 [internal] load metadata for docker.io/library/eclipse-temurin:17-jdk-alpine
#2 ERROR: no match for platform in manifest: not found
------
 > [internal] load metadata for docker.io/library/eclipse-temurin:17-jdk-alpine:
------
Dockerfile:49
--------------------
  47 |     RUN ./docker/copy_jars_from_classpath.sh ${jars_directory}
  48 |     
  49 | >>> FROM eclipse-temurin:17-jdk-alpine AS build_uc
  50 |     
  51 |     ARG unitycatalog_uid
--------------------
ERROR: failed to solve: eclipse-temurin:17-jdk-alpine: failed to resolve source metadata for docker.io/library/eclipse-temurin:17-jdk-alpine: no match for platform in manifest: not found
Returning to base directory /Users/tnightengale/git/unitycatalog/docker/bin
tnightengale@Teghans-14-MBP bin % 
```

1. Run `$ ./build-uc-server-docker` on arm64.

**Expected behavior**

Use a supported jdk and linux distro combination with support for arm64.

**System [please complete the following information]:**

```
tnightengale@Teghans-14-MBP bin % uname -a
Darwin Teghans-14-MBP 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000 arm64
```
**Solutions**

Use a more common distro like Ubuntu for the image. This is probably a good choice for everyone at this point, given that Alpine can [have weird DNS issues](https://martinheinz.dev/blog/92) and anyone who has to login into a box would probably prefer to troubleshoot the service on Debian/ubuntu.

Other solutions could be to use a higher jdk version, which has alpine images for arm64.

**Comment by francisco-ltech** [comment]
Try setting the platform first: 
export DOCKER_DEFAULT_PLATFORM=linux/amd64

Then:
./docker/bin/build-uc-server-docker

**Comment by tnightengale** [comment]
Yes, forcing emulation and using the amd64 image is possible. However, we should aim to support native arm64 images.

**Comment by zcking** [comment]
I do agree @tnightengale emulation is really a workaround, and one that seems to work on M1 and M2 but M3s might be having an issue #470.

The minimal version temurin alpine JDK image I see that supports arm64 is `eclipse-temurin:21-jdk-alpine`.

**Comment by tnightengale** [comment]
@zcking See above PR for new multi-arch image :)
